File: frontend/src/types/modules/report.ts
/**
 * 报表模块类型定义
 */

// 报表查询参数
export interface ReportQuery {
  start_date: string   // YYYY-MM-DD
  end_date: string     // YYYY-MM-DD
  store_id?: number    // 可选门店筛选
  top_n?: number       // TOP N 限制
}

// 日汇总报表行
export interface DailySummaryRow {
  biz_date: string             // 业务日期
  store_id: number             // 门店ID
  store_name: string           // 门店名称
  revenue: number              // 营收
  net_revenue: number          // 净营收
  cost_total: number           // 总成本
  expense_total: number        // 费用总额
  order_count: number          // 订单数
  gross_profit: number         // 毛利
  operating_profit: number     // 营业利润
  gross_profit_rate?: number | null    // 毛利率（%）
  operating_profit_rate?: number | null // 营业利润率（%）
  cost_material: number        // 原材料成本
  cost_labor: number           // 人工成本
  discount_amount: number      // 优惠金额
  refund_amount: number        // 退款金额
}

// 月汇总报表行
export interface MonthlySummaryRow {
  year: number                 // 年份
  month: number                // 月份
  store_id: number             // 门店ID
  store_name: string           // 门店名称
  revenue: number              // 营收
  net_revenue: number          // 净营收
  cost_total: number           // 总成本
  expense_total: number        // 费用总额
  order_count: number          // 订单数
  gross_profit: number         // 毛利
  operating_profit: number     // 营业利润
  gross_profit_rate?: number | null    // 毛利率（%）
  operating_profit_rate?: number | null // 营业利润率（%）
  avg_daily_revenue: number    // 日均营收
  avg_daily_order_count: number // 日均订单数
  day_count: number            // 天数
}

// 门店绩效行
export interface StorePerformanceRow {
  store_id: number             // 门店ID
  store_name: string           // 门店名称
  revenue: number              // 营收
  net_revenue: number          // 净营收
  order_count: number          // 订单数
  avg_order_amount: number     // 平均订单金额
  gross_profit: number         // 毛利
  operating_profit: number     // 营业利润
  gross_profit_rate?: number | null    // 毛利率（%）
  operating_profit_rate?: number | null // 营业利润率（%）
  revenue_rank: number         // 营收排名
  profit_rank: number          // 利润排名
}

// 费用明细行
export interface ExpenseBreakdownRow {
  expense_type_id: number      // 费用科目ID
  type_code: string            // 科目代码
  type_name: string            // 科目名称
  category: string             // 费用类别
  total_amount: number         // 总金额
  record_count: number         // 记录数
  avg_amount: number           // 平均金额
  percentage: number           // 占比（%）
}

// 报表响应数据
export interface ReportResponse<T> {
  code: number
  message: string
  data: T[]
}

File: backend/app/schemas/report.py
"""
报表模块 Schema 定义
"""
from datetime import date
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel, Field


# ==================== 查询参数 ====================

class ReportQuery(BaseModel):
    """报表查询参数"""
    start_date: date = Field(..., description="开始日期")
    end_date: date = Field(..., description="结束日期")
    store_id: Optional[int] = Field(None, description="门店ID（为空表示全部门店）")
    top_n: Optional[int] = Field(None, ge=1, le=100, description="TOP N排名（1-100）")
    group_by: Optional[str] = Field(None, description="分组字段（day/week/month/store）")


# ==================== 响应模型 ====================

class DailySummaryRow(BaseModel):
    """日汇总行"""
    biz_date: date = Field(..., description="业务日期")
    store_id: Optional[int] = Field(None, description="门店ID")
    store_name: Optional[str] = Field(None, description="门店名称")
    
    # 营收指标
    revenue: Decimal = Field(default=Decimal("0.00"), description="营业收入")
    net_revenue: Decimal = Field(default=Decimal("0.00"), description="净收入")
    discount_amount: Decimal = Field(default=Decimal("0.00"), description="优惠金额")
    refund_amount: Decimal = Field(default=Decimal("0.00"), description="退款金额")
    
    # 成本指标
    cost_total: Decimal = Field(default=Decimal("0.00"), description="总成本")
    cost_material: Decimal = Field(default=Decimal("0.00"), description="原材料成本")
    cost_labor: Decimal = Field(default=Decimal("0.00"), description="人工成本")
    
    # 费用指标
    expense_total: Decimal = Field(default=Decimal("0.00"), description="总费用")
    
    # 订单指标
    order_count: int = Field(default=0, description="订单数")
    
    # 利润指标
    gross_profit: Decimal = Field(default=Decimal("0.00"), description="毛利润")
    operating_profit: Decimal = Field(default=Decimal("0.00"), description="营业利润")
    gross_profit_rate: Optional[Decimal] = Field(None, description="毛利率（%）")
    operating_profit_rate: Optional[Decimal] = Field(None, description="营业利润率（%）")
    
    class Config:
        from_attributes = True


class MonthlySummaryRow(BaseModel):
    """月汇总行"""
    year: int = Field(..., description="年份")
    month: int = Field(..., description="月份")
    store_id: Optional[int] = Field(None, description="门店ID")
    store_name: Optional[str] = Field(None, description="门店名称")
    
    # 营收指标
    revenue: Decimal = Field(default=Decimal("0.00"), description="营业收入")
    net_revenue: Decimal = Field(default=Decimal("0.00"), description="净收入")
    discount_amount: Decimal = Field(default=Decimal("0.00"), description="优惠金额")
    refund_amount: Decimal = Field(default=Decimal("0.00"), description="退款金额")
    
    # 成本指标
    cost_total: Decimal = Field(default=Decimal("0.00"), description="总成本")
    
    # 费用指标
    expense_total: Decimal = Field(default=Decimal("0.00"), description="总费用")
    
    # 订单指标
    order_count: int = Field(default=0, description="订单数")
    
    # 利润指标
    gross_profit: Decimal = Field(default=Decimal("0.00"), description="毛利润")
    operating_profit: Decimal = Field(default=Decimal("0.00"), description="营业利润")
    gross_profit_rate: Optional[Decimal] = Field(None, description="毛利率（%）")
    operating_profit_rate: Optional[Decimal] = Field(None, description="营业利润率（%）")
    
    # 日均指标
    avg_daily_revenue: Decimal = Field(default=Decimal("0.00"), description="日均收入")
    avg_daily_order_count: Decimal = Field(default=Decimal("0.00"), description="日均订单数")
    
    class Config:
        from_attributes = True


class StorePerformanceRow(BaseModel):
    """门店绩效行"""
    store_id: int = Field(..., description="门店ID")
    store_name: str = Field(..., description="门店名称")
    
    # 营收指标
    revenue: Decimal = Field(default=Decimal("0.00"), description="营业收入")
    net_revenue: Decimal = Field(default=Decimal("0.00"), description="净收入")
    
    # 订单指标
    order_count: int = Field(default=0, description="订单数")
    avg_order_amount: Decimal = Field(default=Decimal("0.00"), description="客单价")
    
    # 利润指标
    gross_profit: Decimal = Field(default=Decimal("0.00"), description="毛利润")
    operating_profit: Decimal = Field(default=Decimal("0.00"), description="营业利润")
    gross_profit_rate: Optional[Decimal] = Field(None, description="毛利率（%）")
    operating_profit_rate: Optional[Decimal] = Field(None, description="营业利润率（%）")
    
    # 排名
    revenue_rank: Optional[int] = Field(None, description="营收排名")
    profit_rank: Optional[int] = Field(None, description="利润排名")
    
    class Config:
        from_attributes = True


class ExpenseBreakdownRow(BaseModel):
    """费用明细行"""
    expense_type_id: int = Field(..., description="费用科目ID")
    expense_type_code: str = Field(..., description="费用科目编码")
    expense_type_name: str = Field(..., description="费用科目名称")
    category: str = Field(..., description="费用类别")
    
    # 门店信息（可选）
    store_id: Optional[int] = Field(None, description="门店ID")
    store_name: Optional[str] = Field(None, description="门店名称")
    
    # 费用统计
    total_amount: Decimal = Field(default=Decimal("0.00"), description="费用总额")
    record_count: int = Field(default=0, description="记录笔数")
    avg_amount: Decimal = Field(default=Decimal("0.00"), description="平均单笔金额")
    
    # 占比
    percentage: Optional[Decimal] = Field(None, description="占总费用比例（%）")
    
    class Config:
        from_attributes = True
